nav2-usability/scenario-code/deeplink-queryparam/deeplink_queryparam_vrouter.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Deeplink query parameters example
+/// Done using VRouter
+
+import 'package:flutter/material.dart';
+import 'package:vrouter/vrouter.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+class Book {
+  final String title;
+  final String author;
+
+  Book(this.title, this.author);
+}
+
+class BooksApp extends StatelessWidget {
+  final List<Book> books = [
+    Book('Stranger in a Strange Land', 'Robert A. Heinlein'),
+    Book('Foundation', 'Isaac Asimov'),
+    Book('Fahrenheit 451', 'Ray Bradbury'),
+  ];
+
+  @override
+  Widget build(BuildContext context) {
+    return VRouter(
+      routes: [
+        VWidget(path: '/', widget: BooksListScreen(books: books)),
+      ],
+    );
+  }
+}
+
+class BooksListScreen extends StatelessWidget {
+  final List<Book> books;
+
+  BooksListScreen({required this.books});
+
+  @override
+  Widget build(BuildContext context) {
+    final filter = context.vRouter.queryParameters['filter'];
+    return Scaffold(
+      appBar: AppBar(),
+      body: ListView(
+        children: [
+          TextField(
+            decoration: InputDecoration(
+              hintText: 'filter',
+            ),
+            onSubmitted: (value) =>
```

Comment:
Does this push a new page on top? If so, I think we should do replace or just update the current page instead

Reply:
This does what you want. It just updates the current page.

Reply:
I understand the confusion, I used push because people are used to it. In VRouter you can't really "push on top" anyway.